ludwig/utils/commit_utils.py

```diff
@@ -0,0 +1,109 @@
+                the models weights as well as other associated training artifacts
+                are saved.
+            repo_type (`str`, *optional*):
+                Set to `"dataset"` or `"space"` if uploading to a dataset or
+                space, `None` or `"model"` if uploading to a model. Default is
+                `None`.
+            private (`bool`, *optional*, defaults to `False`):
+                Whether the model repo should be private.
+            commit_message (`str`, *optional*):
+                The summary / title / first line of the generated commit. Defaults to:
+                `f"Upload {path_in_repo} with huggingface_hub"`
+            commit_description (`str` *optional*):
+                The description of the generated commit
+        """
+        # Validate repo_id has both a namespace and a repo name
+        assert "/" in repo_id, (
+            "`repo_id` must be a namespace (user or an organization) and a repo name separated by a `/`."
+            " For example, if your HF username is `johndoe` and you want to create a repository called `test`, the"
+            " repo_id should be johndoe/test"
+        )
+
+        # Make sure the model's save path is actually a valid path
+        if not os.path.exists(model_path):
+            raise FileNotFoundError(f"The path '{model_path}' does not exist.")
+
+        # Make sure the model is actually trained
+        trained_model_artifacts_path = os.path.join(model_path, "model", "model_weights")
+        if not os.path.exists(trained_model_artifacts_path):
+            raise Exception(
+                f"Model artifacts not found at {trained_model_artifacts_path}. "
+                f"It is possible that model at '{model_path}' hasn't been trained yet, or something went"
+                "wrong during training where the model's weights were not saved."
+            )
+
+        # Make sure the model's saved artifacts either contain:
+        # 1. pytorch_model.bin -> regular model training, such as ECD or for LLMs
+        # 2. adapter_model.bin -> LLM fine-tuning using PEFT
+        files = set(os.listdir(trained_model_artifacts_path))
+        if "pytorch_model.bin" not in files and "adapter_model.bin" not in files:
+            raise Exception(
+                f"Can't find model weights at {trained_model_artifacts_path}. Trained model weights should "
+                "either be saved as `pytorch_model.bin` for regular model training, or have `adapter_model.bin`"
+                "if using parameter efficient fine-tuning methods like LoRA."
+            )
+
+        # Create empty model repo using repo_id, but it is okay if it already exists.
+        self.api.create_repo(
+            repo_id=repo_id,
+            private=private,
+            repo_type=repo_type,
+            exist_ok=True,
+        )
+
+        # Upload all artifacts in model weights folder
+        upload_path = self.api.upload_folder(
```

Comment:
should we return a status from this method indicating the different levels of success

Reply:
Yeah we can do that, let me follow up